src/shutdown.rs

```diff
@@ -103,11 +116,9 @@ impl SourceShutdownCoordinator {
         let shutdown_signal =
             ShutdownSignal::new(shutdown_begun_tripwire, shutdown_complete_trigger);
 
-        // shutdown_source_end drops the force_shutdown_trigger even on success when we should *not*
-        // be shutting down.  Dropping the trigger will cause the Tripwire to resolve with an error,
-        // so we use or_else with future::empty() to make it so it never resolves if the Trigger is
-        // prematurely dropped instead.
-        let force_shutdown_tripwire = force_shutdown_tripwire.or_else(|_| future::empty());
+        // `force_shutdown_tripwire` resolved even on success when we should *not* be shutting down.
+        // `tripwire_handler` will check resolved value and never resolve future if this not required.
+        let force_shutdown_tripwire = force_shutdown_tripwire.then(tripwire_handler);
```

Comment:
This documentation should be updated to something like:
```
`force_shutdown_tripwire` resolves even if canceled when we should *not* be shutting down. 
`tripwire_handler` handles cancel by never resolving.
```

Noting that this is has different behavior than the old code, but it's usage checks out.